### PR TITLE
Added N-Queens Problem in C++

### LINKED ISSUE #951: N-Queen Problem in c++

### PR DESCRIPTION
**Pull request by somya-kapoor**
Fixes #951 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added N-Queens Problem in C++

#### Languages Used:
- C++

#### Files Added:
- Queens_Problem.cpp

### CODE REVIEW

**Comment by MastersAbh** [comment]
For input n=2 and n=3, the code prints an empty chessboard, instead add conditions to justify why is it empty and the queens cannot be placed.

**Comment by somya-kapoor** [comment]
@MastersAbh I have done the requires changes.Please review

**Comment by MastersAbh** [comment]
I still get an empty chessboard on inputs <= 3

**Comment by somya-kapoor** [comment]
@MastersAbh  Sorry by mistake I made if statement till 2 only but I have changed it to 3 . Please review

**Comment by MastersAbh** [comment]
But i am still seeing empty matrix in output. Please remove that.

**Comment by somya-kapoor** [comment]
@MastersAbh  I have done the required changes. Please review.

File: Queens_Problem/Queens_Problem.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+int main()
+{
+    cout << "Enter the value of N for NxN chessboard : ";
+    cin >> N;
+
+    int row, col;
+    if ( N <= 3)
+    {   
+        cout<< "The Queens cant be placed in the chessboard because queens attack can each other \n" ;
```
Comment (SakshiShreya) [inline Queens_Problem/Queens_Problem.cpp]: Please change the line to `... queens can attack each other`

File: Queens_Problem/Queens_Problem.cpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        for( col = 0 ; col < N ; col++ )
+        {
+            //checking if we can place a queen here or not
+            //queen will not be placed if the place is not safe
+            //or already occupied
+            if((is_safe(row,col)) && (board[row][col]!=1))
+            {
+                board[row][col] = 1;
+                //recursion;wether we can put the next queen with this arrangment or not
+                if(N_queen( n-1 ) == 1)
```
Comment (SakshiShreya) [inline Queens_Problem/Queens_Problem.cpp]: Fix spacing in line 73.

File: Queens_Problem/Queens_Problem.cpp [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+using namespace std;
+
+int N;
+int board[100][100];
+int is_safe(int row, int col);
+int N_queen(int n);
+
+int main()
+{
+    cout << "Enter the value of N for NxN chessboard : ";
```
Comment (SakshiShreya) [inline Queens_Problem/Queens_Problem.cpp]: Remove space before `:`.

File: Queens_Problem/Queens_Problem.cpp [hunk 4]
```diff
@@ -69,7 +69,7 @@ int N_queen(int n)
             if((is_safe(row,col)) && (board[row][col]!=1))
             {
                 board[row][col] = 1;
-                //recursion;wether we can put the next queen with this arrangment or not
+                //recursion ; whether we can put the next queen with this arrangement or not
                 if(N_queen( n-1 ) == 1)
```
Comment (SakshiShreya) [inline Queens_Problem/Queens_Problem.cpp]: Line 72: Remove the space before `;`.

Line 73: spacing is not fixed yet - `if(N_queen( n-1 ) == 1)` should be `if(N_queen(n - 1) == 1)`.

 We never add space before punctuation marks. Neither in english, nor in code.
We never add space just after an opening bracket or just before a closing bracket both in english and code.
We should try to always add a space before and after operators in code. (in english also)

**Comment by somya-kapoor** [comment]
@SakshiShreya I have done the required changes. Please review.

**Pull request merged**